### PR TITLE
[CARBONDATA-2984][Streaming] Fix NPE when there is no data in the task of a batch

### PR DESCRIPTION
**Pull request by QiangCai**
Fix NPE when there is no data in the task of a batch

Streaming batch maybe has no data, so it doesn't require to append blocklet to streaming file. So it doesn't need to update min/max index of streaming file, just use min/max index of old file .

 - [x] Any interfaces changed?
 no
 - [x] Any backward compatibility impacted?
 no
 - [x] Document update required?
 no
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
           add test case
        - How it is tested? Please attach test report.
        - Is it a performance-related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
small changes

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/625/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/817/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8886/

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala [hunk 1]
```diff
@@ -2327,7 +2371,7 @@ class TestStreamingTableOperation extends QueryTest with BeforeAndAfterAll {
             .load()
 
           // Write data from socket stream to carbondata file
-          qry = readSocketDF.writeStream
+          qry = readSocketDF.repartition(2).writeStream
```
Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: Is this the place to generate empty rows to insert to streaming file?

**Comment by xuchuanyin** [comment]
the testcases in `TestStreamingTableOperation.scala` are tight independent with each other, I suggest you to refactor before adding more tests.

**Comment by QiangCai** [comment]
@xuchuanyin different test cases use different tables, they are independent.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/841/

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala (hunk 1 above)
Comment (QiangCai) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: fixed

**Comment by zzcclp** [comment]
can this pr already be merged?

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**